src/components/Charts/utils.ts

```diff
@@ -0,0 +1,172 @@
+import _ from 'lodash';
+import Highcharts, { dateFormat } from 'highcharts';
+import palette from '../../assets/theme/palette';
+
+export const last = (list: any[]) => list[list.length - 1];
+
+export const formatDecimal = (num: number, places = 2): string =>
+  num.toFixed(places);
+
+export const formatPercent = (num: number): string =>
+  `${formatDecimal(100 * num, 1)} %`;
+
+export const parseDate = (date: Date): number => new Date(date).valueOf();
+
+export const currentValueAnnotation = (
+  x: number,
+  y: number,
+  text: string,
+): Highcharts.AnnotationsOptions => ({
+  labelOptions: {
+    backgroundColor: palette.white,
+  },
+  labels: [
+    {
+      align: 'left',
+      // @ts-ignore - Bug in Highchart types
+      verticalAlign: 'center',
+      x: 12,
+      shape: 'rect',
+      point: {
+        xAxis: 0,
+        yAxis: 0,
+        x,
+        y,
+      },
+      style: {
+        fontSize: '14px',
+        fontWeight: 'bold',
+      },
+      text,
+      className: 'ZoneAnnotation ZoneAnnotation--CurrentValue',
+    },
+  ],
+});
+
+export const getTickPositions = (
+  minY: number,
+  maxY: number,
+  zones: Zone[],
+): (number | undefined)[] =>
+  zones
+    .map(zone => zone.value)
+    .filter(val => !_.isUndefined(val))
+    .concat([minY, maxY])
+    .sort();
+
+export const getMaxY = (data: Highcharts.Point[]) => _.max(data.map(d => d.y));
+
+export const roundAxisLimits = (axisMin: number, axisMax: number) => [
+  axisMin,
+  _.ceil(1.2 * axisMax, 1),
+];
+
+export const baseOptions: Highcharts.Options = {
+  title: {
+    text: undefined,
+  },
+  subtitle: {
+    text: undefined,
+  },
+  plotOptions: {
+    series: {
+      animation: false,
+      marker: {
+        enabled: false,
+      },
+    },
+  },
+  legend: {
+    enabled: false,
+  },
+  xAxis: {
+    type: 'datetime',
+    // @ts-ignore - Bug in Highchart types
+    step: 7,
+    labels: {
+      useHTML: true,
+      rotation: 0,
+      formatter: function () {
+        return dateFormat('%b %e', this.value);
+      },
+    },
+  },
+  yAxis: {
+    title: {
+      text: undefined,
+    },
+    labels: {
+      useHTML: true,
+      formatter: function () {
+        return this.value === 0
+          ? ''
+          : this.axis.defaultLabelFormatter.call(this);
+      },
+    },
+    gridLineDashStyle: 'Dash',
+  },
+};
+
+const annotationZoneLabelBase: Highcharts.AnnotationsLabelOptions = {
+  style: {
+    fontSize: '14px',
+    fontWeight: 'bold',
+  },
+  align: 'right',
+  // @ts-ignore - Bug in Highchart types
+  verticalAlign: 'center',
+  x: 12,
+  shape: 'rect',
+};
+
+interface Zone {
+  className: string;
+  color: string;
+  labelClassName: string;
+  name: string;
+  value?: number;
+}
+
```

Comment:
oh what is the difference for this and the one in the enum

Reply:
This one is what the chart needs to color the line by zones, we could rename `label` → `name` in the enum to avoid an unnecessary transformation, but the rest is necessary (we could rename `upperLimit` → `value`, but I think that's confusing, so I would prefer to keep it as `upperLimit`).